drivers/Aqara/aqara-presence-sensor/src/fp2/discovery_helper.lua

```diff
@@ -0,0 +1,79 @@
+local log = require "log"
+local discovery_helper = {}
+
+local SERVICE_TYPE = "_Aqara-FP2._tcp"
+local DOMAIN = "local"
+
+local fp2_api = require "fp2.api"
+local discovery_mdns = require "discovery_mdns"
+
+function discovery_helper.get_dni(driver, ip, discovery_responses)
+  local text_list = discovery_mdns.find_text_list_in_mdns_response(driver, ip, discovery_responses)
+  for _, text in ipairs(text_list) do
+    for key, value in string.gmatch(text, "(%S+)=(%S+)") do
+      if key == "mac" then
+        return value
+      end
+    end
+  end
+
+  log.error("discovery_helper.get_dni : failed to find dni")
+  return nil
+end
+
+function discovery_helper.get_service_type_and_domain()
+  return SERVICE_TYPE, DOMAIN
+end
+
+function discovery_helper.get_device_create_msg(driver, device_dni, device_ip)
+  local device_info = fp2_api.get_info(device_ip, fp2_api.labeled_socket_builder(device_dni))
+
+  if not device_info then
+    log.error(string.format("failed to create device create msg. device_info is nil. dni = %s"))
```

Comment:
This is attempting to format a string with an argument but the argument isn't provided

Reply:
removed the argument because it is nil